src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -220,15 +233,24 @@ ClientRequest::do_process(Ref<PG>& pg, crimson::osd::ObjectContextRef obc)
       return seastar::make_ready_future<seq_mode_t>(seq_mode_t::OUT_OF_ORDER);
     } else if (const hobject_t& hoid = m->get_hobj();
                !pg->get_peering_state().can_serve_replica_read(hoid)) {
-      auto reply = crimson::make_message<MOSDOpReply>(
-	m.get(), -EAGAIN, pg->get_osdmap_epoch(),
-	m->get_flags() & (CEPH_OSD_FLAG_ACK|CEPH_OSD_FLAG_ONDISK),
-	!m->has_flag(CEPH_OSD_FLAG_RETURNVEC));
-      return conn->send(std::move(reply)).then([] {
-        return seastar::make_ready_future<seq_mode_t>(seq_mode_t::OUT_OF_ORDER);
-      });
+      return reply_op_error(pg, -EAGAIN);
     }
   }
+  if (m->get_oid().name.size()
+      > crimson::common::local_conf()->osd_max_object_name_len) {
+    return reply_op_error(pg, -ENAMETOOLONG);
+  }
+  if (m->get_hobj().get_key().size()
+      > crimson::common::local_conf()->osd_max_object_name_len) {
+    return reply_op_error(pg, -ENAMETOOLONG);
+  }
+  if (m->get_hobj().nspace.size()
+      > crimson::common::local_conf()->osd_max_object_namespace_len) {
+    return reply_op_error(pg, -ENAMETOOLONG);
+  }
+  if (m->get_hobj().oid.name.empty()) {
+    return reply_op_error(pg, -EINVAL);
+  }
```

Comment:
How about dissecting these sanity checks into a dedicated method?

Reply:
I keep it as it, it hardly handle the continuation logic.